actionview/lib/action_view/helpers/form_with_helper.rb

```diff
@@ -0,0 +1,196 @@
+          if include_hidden
+            hidden_options = options.merge(type: 'hidden', value: off)
+            hidden_options.delete(:checked)
+            hidden = tag.input(hidden_options).html_safe
+          end
+          hidden + tag.input(checkbox_options).html_safe
+        end
+
+        def select(method, choices = nil, blank: nil, prompt: nil, index: :undefined, disabled: nil, **options, &block)
+          tag.select option_tags_for_select(choices, blank: blank), options_for(method, options)
+        end
+
+        def collection_select(method, collection, value_method, text_method, blank: nil, prompt: nil, index: :undefined, disabled: nil, **options)
+          choices = collection.map do |object|
+            [object.send(value_method), object.send(text_method)]
+          end
+          select(method, choices, options)
+        end
+
+        def option_tags_for_select(choices, blank: false)
+          result = (blank ? tag.option("", value: "") : "").html_safe
+          result += choices.map do |choice|
+            if choice.is_a? Array
+              tag.option choice[0], value: choice[1]
+            else
+              tag.option choice, value: choice
+            end
+          end.join("\n").html_safe
+          result
+        end
+
+        private
+          def checked?(on_value, value)
+            case value
+            when TrueClass, FalseClass
+              value == !!on_value
+            when NilClass
+              false
+            when String
+              value == on_value
+            else
+              if value.respond_to?(:include?)
+                value.include?(on_value)
+              else
+                value.to_i == on_value.to_i
+              end
+            end
+          end
+
+          def translate_with_human_attribute_name(method)
+            model && model.class.respond_to?(:human_attribute_name) ? model.class.human_attribute_name(method) : nil
+          end
+
+          def submit_default_value
+            if scope
+              key = model ? (model.persisted? ? :update : :create) : :submit
+              model_name = model.respond_to?(:model_name) ? model.model_name.human : scope.to_s.humanize
+              defaults = [:"helpers.submit.#{scope}.#{key}", :"helpers.submit.#{key}", :"#{key.to_s.humanize} #{model_name}"]
+              I18n.t(defaults.shift, model: model_name, default: defaults)
+            else
+              I18n.t("helpers.submit.no_model")
+            end
+          end
+
+          def name_for(method, options)
+            scope = options.delete(:scope) { @scope }
+            index = options.delete(:index)
+            method = method.to_s.chomp("?")
+            name = scope.to_s
+            name += index ? "[#{index}]" : ""
+            name += scope.nil? ? method : "[#{method}]"
+            name += options.delete(:multiple) ? "[]" : "" #exception => file_field
+          end
+
+          def id_for(method, scope, **options)
```

Comment:
This doesn't seem to be used? Besides I think `dom_id(model)` is what's to be used for this.